### ISSUE TITLE
Infinite "loop" in MessagesManager.cpp

### ISSUE THREAD

**Issue by Cavallium** [tdlib/td#1362]
In MessagesManager.cpp, add_pending_update started to get called forever, and postponed pts size growed until filling the memory almost immediatly. TDLib stopped responding.
The source of these infinite "add_pending_update" calls is after_get_difference().
Unfortunately it happened to me only once, so I can't reproduce it.
I am using the latest commit of tdlib.

**Comment by levlam** [comment]
Do you have full trace?

**Comment by Cavallium** [comment]
Unfortunately not.
Is it possible that this change caused this?
commit 3dbc996accf5c7e12c338b4bd31c0cef0ed68f15
MessagesManager.cpp, line 6239

In the old code, the function "after_get_difference()" loops all "postponed_pts_updates_", and in the loop calls "add_pending_update()", that re-adds them in "postponed_pts_updates_" if running_get_difference is true.
In the new code "add_pending_update()" readds them also if "postponed_pts_updates_" is not empty, which is true since we are calling the method from inside the iteration of that collection.

**Comment by levlam** [comment]
There was a lot of changes in update handling recently, which could cause this, but you are likely to find the right commit.

**Comment by levlam** [comment]
This is likely to be fixed in https://github.com/tdlib/td/commit/eab5532688211cc6dfd6806efa4ac9b0ae49d4c7.
